client/search/search-input.tsx

```diff
@@ -1,54 +1,35 @@
-import React, { useCallback, useImperativeHandle, useRef, useState } from 'react'
+import React, { useCallback, useState } from 'react'
 import styled from 'styled-components'
 import SearchIcon from '../icons/material/baseline-search-24px.svg'
 import { useKeyListener } from '../keyboard/key-listener'
 import { fastOutSlowInShort } from '../material/curves'
-import TextField from '../material/text-field'
-import { useValueAsRef } from '../state-hooks'
+import { TextField, TextFieldHandle } from '../material/text-field'
+import { useMultiRef } from '../state-hooks'
 
 const ESCAPE = 'Escape'
 const F = 'KeyF'
 
 const TextFieldContainer = styled(TextField)`
-  width: ${props =>
-    props.isFocused
-      ? 'var(--sb-search-input-focused-width, 250px)'
-      : 'var(--sb-search-input-width, 200px)'};
-
+  width: var(--sb-search-input-width, 200px);
```

Comment:
What I was saying in Discord was, you don't need to have these custom CSS props at all any more, just put this styling in the places that are using this thing and use the static width value there. (I don't honestly know that this component needs to exist at all either, and having a generic component that exists to eat particular keypresses on the page feels a bit odd to me. That feels like it's more the concern of the page itself and would be better handled at that level, but, up to you I guess)

Reply:
And while we're here, 250 is not divisible by 4 and thus isn't a valid width, you probably want 256 or something

Reply:
Yeah, my initial thinking was to have this component to encapsulate keyboard handling (Ctrl + F to focus search, Escape to unfocus) and potentially show a spinner somewhere next to it. But without a spinner, all this component does is handle these two hotkeys then. I could stop eating the Escape hotkey by tracking the focus state again as I was doing and only handle it if the input is focused. But that still leaves the Ctrl + F hotkey which, as you said, is probably the concern of the page itself since there could be more than one search inputs on a page. Hmm.

I'll remove it :d